.travis.yml

```diff
@@ -19,32 +19,42 @@ jobs:
       - stage:   build
         env:     PROG=gocsi.a
         install: true
```

Comment:
Hi @sbezverk,

Perhaps I misspoke earlier. The intent was not for you to modify the Travis-CI configuration, but for you to run `dep ensure` locally to update the CSI dependency to `master`. Then you could commit the updated dependency along with the rest of your changes.

Reply:
@akutz here is the thing, I added ```make dep-ensure``` not just for fun but because I discovered that CI does not update vendor folder and I can easily prove it to you. Not sure where there is a bug. I think it makes sense to update dependency before running building, as you: 1 - clone from master so the old vendor get copied, 2 - you do not allow PR to have updated vendor folder.  You see the problem right?

Reply:
HI @sbezverk,

I think you misunderstand the intent. The CI is not supposed to update the vendor folder for GoCSI. The CI workflow **only** runs `dep ensure` for the generated SP via the `make csi-sp` build step. It is the responsibility of the developer to run `dep ensure` locally and commit any changes to the local vendor directory.

A few notes:

* The `./vendor` directory at the root of GoCSI is to be updated locally and committed along with any changes to GoCSI.
* The SP generated by `make csi-sp` or `./gocsi.sh` (the former calls the latter) includes a new `Gopkg.toml` file. Try it locally. Run `./gocsi.sh csi-sp`. It will create sources at `$GOPATH/src/csi-sp`, including a new `Gopkg.toml` file. This example and process are both documented in [Bootstrapping a Storage Plug-in](https://github.com/thecodeteam/gocsi#bootstrapping-a-storage-plug-in).
* I hear you that some CI processes update dependencies as part of the CI process, but ours does not. GoCSI, like REX-Ray, commit the entirety of vendor so that someone can simply do `go get github.com/thecodeteam/rexray` or `go get github.com/thecodeteam/gocsi/csc` and those two will build successfully because their vendored dependencies are committed alongside their sources.

Please let me know if you have any more questions.

Reply:
Hi @sbezverk,

I'm responding to your note below about including your own vendor files. All PRs that have updated dependencies should run `dep ensure` and commit any changes to GoCSI `./vendor`. My remark about not including your own vendor files in a PR was regarding you not modifying those files **manually**, as you did with `csi.pb.go`. What I expected and we require was for you to update GoCSI's `Gopkg.toml` and point it at a fork or ref that includes the upstream changes to the CSI spec. I think that statement must have set you on the path of believing that no changes in `vendor` were allowed when what I said was no **manual** changes -- only those as a result of `dep ensure`.

Reply:
@akutz I knew I missed something. Thank you for clarifying everything! I will change PR as per your last comment and will push it for your review.